crawl-ref/source/cloud.cc

```diff
@@ -1814,3 +1816,105 @@ void end_still_winds()
+            if (special == special_fog_type::normal) {
+                int runs = 1;
+                if (type == CLOUD_SALT) //Desolation
+                    runs = 20;
+                while (delay <= 0)
+                    run(runs);
+            }
+            if (special == special_fog_type::chained) {
+                if (env.special_clouds == 0) env.special_clouds = 1;
+                //Find and store pointers to members of the chain, only once.
+                if (slave_references.empty() && slaves.size() != 0) {
+                    for (CloudGenerator &generator : env.cloud_generators)
+                        for (int i = 0; i < slaves.size(); i++) {
+                            if (generator.marker_id == slaves[i] &&
+                                generator.marker_id != marker_id) //Don't slave self.
+                            {
+                                slave_references.push_back(&generator);
+                                // End early if all slaves are found.
+                                if (slaves.size() - 1 == slave_references.size())
+                                    break;
+                            }
+                        }
+                }
+                //Just in case someone makes a vault with a single chained fog machine
+                if (!slave_references.empty()) {
+                    for (CloudGenerator *generator : slave_references) {
+                        generator->buildup_turns += you.time_taken;
+                        //Set this here instead of when they're actually run
+                        //so grouped clouds run on a single master's delay.
+                        //(i.e., their delays won't be checked, even if)
+                        generator->touched = 1;
+                    }
+                    //If the master's delay is 0, run all chained generators.
+                    while (delay <= 0)
+                    {
+                        run(1);
+                        for (CloudGenerator *generator : slave_references)
+                            generator->run(1);
+                    }
+                }
+            }
+        }
+    }
+}
+
+void CloudGenerator::run(int runs)
+{
+    if (loc.distance_from(you.pos()) <= 10) {
+        for (int i = 0; i < runs; i++) {
+            const coord_def centre = random_walk(loc, walk_dist);
+            const pair<int, int> size_range = get_size_range();
+            int size = random_range(size_range.first, size_range.second);
+            int spread = get_spread();
+
+            apply_area_cloud(
+                [this](coord_def where, int /*pow*/, int spread,
+                    cloud_type /*type*/, const actor* agent, int excl_rad)
+            {
+                const int pow = random_range(pow_min, pow_max, pow_rolls);
+                place_cloud(type, where, pow, agent, spread, excl_rad);
+                return 1;
+            }, centre, 0, size, type, 0, spread, exclusion_radius);
+
```

Comment:
This lambda expression is one place where braces on their own lines isn't required.